bin/main.rb

```diff
@@ -1,4 +1,141 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
-puts 'Hello world!'
+require 'io/console'
+
+class Game
+  def initialize(board)
+    @game_board = board
+
+    @game_board.ask_names
+    @game_board.draw
+  end
+
+  def start
+    while @game_board.count_empties.positive? && !@game_board.winner
+      @game_board.mark
+      @game_board.draw
+    end
+
+    if @game_board.winner
+      puts "#{@game_board.winner.name} wins!!"
+    else
+      puts 'this is a draw'
+    end
+  end
+end
+
+class Board
+  attr_reader :winner
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @blocks = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+    @flag = true
+
+    @winner_patterns = []
+    @winner_patterns.push([1, 1, 1, 0, 0, 0, 0, 0, 0])
+    @winner_patterns.push([0, 0, 0, 1, 1, 1, 0, 0, 0])
+    @winner_patterns.push([0, 0, 0, 0, 0, 0, 1, 1, 1])
+    @winner_patterns.push([1, 0, 0, 1, 0, 0, 1, 0, 0])
+    @winner_patterns.push([0, 1, 0, 0, 1, 0, 0, 1, 0])
+    @winner_patterns.push([0, 0, 1, 0, 0, 1, 0, 0, 1])
+    @winner_patterns.push([1, 0, 0, 0, 1, 0, 0, 0, 1])
+    @winner_patterns.push([0, 0, 1, 0, 1, 0, 1, 0, 0])
+
```

Comment:
This is really hardcoded. For the next milestone it will be important to not have this code so wet (verbose).

Reply:
@winner_patterns = [
      [1, 1, 1, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 1, 1, 1, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 1, 1, 1],
      [1, 0, 0, 1, 0, 0, 1, 0, 0],
      [0, 1, 0, 0, 1, 0, 0, 1, 0],
      [0, 0, 1, 0, 0, 1, 0, 0, 1],
      [1, 0, 0, 0, 1, 0, 0, 0, 1],
      [0, 0, 1, 0, 1, 0, 1, 0, 0]
    ]

Reply:
1 | 2 | 3        X | O | O 
-----------      -----------
 4 | 5 | 6   =>   X | O | X 
-----------      -----------
 7 | 8 | 9        X | X | O 
============================
yochi wins!!

Reply:
```suggestion
 1 | 2 | 3        X | O | O 
-----------      -----------
 4 | 5 | 6   =>   X | O | X 
-----------      -----------
 7 | 8 | 9        X | X | O 
============================
yochi wins!!
```

Reply:
```
 1 | 2 | 3        X | O | O 
-----------      -----------
 4 | 5 | 6   =>   X | O | X 
-----------      -----------
 7 | 8 | 9        X | X | O 
============================
yochi wins!!

```

Reply:
```
 1 | 2 | 3        X | O | X 
-----------      -----------
 4 | 5 | 6   =>   O | X | O 
-----------      -----------
 7 | 8 | 9        X |   |   
============================
yochi wins!!
```

Reply:
```suggestion
    @winner_patterns.any? do |i|
      r = i.each_with_index.inject(0) do |acc, (val, index)|
        val == arr[index] && val == 1 ? acc + 1 : acc
      end
      r == 3
    end
```